### ISSUE TITLE
Update CDI Extension: Hikari CP documentation autoCommit: false; Updating Title: JPA Transaction is not rolling back with rollBackOn is set and Exception thrown

### ISSUE THREAD

**Issue by nagaraja0yellapu** [oracle/helidon#1176]
## Environment Details
* Helidon Version: 1.4.0
* Helidon MP
* JDK version: 1.8
* OS: Windows 10
* Docker version (if applicable):

----------

## Problem Description
[//]: # "Describe the bug in detail highlighting current behavior vs expected behavior"
[//]: # "State if the problem is easily reproducible or happens intermittently"
[//]: # "Include stack traces or command outputs"

The complete transaction has to be rolledback in below case if BadRequestException is thrown
```
@Transactional(value = Transactional.TxType.REQUIRES_NEW, rollbackOn = BadRequestException.class)
public Response save(Role model) {
            Role origModel = repository.findOne(model.getId());
            origModel.setDescription("testing456");
            throw new BadRequestException("to test rollback");
}
```

But due to HikariCP default property **autoCommit: true**, the transaction is committed, even though Exception is thrown. Ref for autoCommit property: https://github.com/brettwooldridge/HikariCP/blob/dev/README.md#configuration-knobs-baby

In the real world scenario & as per JTA specification, we need the full transaction to be rolledback on throwing corresponding Exception. It would be nice to see to have this information in **Helidon CDI extension for HikariCP** Documentation as below
```
javax.sql.DataSource.orders.autoCommit = false
```

**Comment by nagaraja0yellapu** [comment]
After trying with **javax.sql.DataSource.orders.autoCommit = false**, I found that the data is not saving in database even though there is no Exception. This needs to be addressed properly. With hands on project development with Helidon, I observed that we can't use Helidon JPA, JTA & Hikari CP at current state for production. For now, I am switching back to Helidon MP + Eclipselink with RESOURCE_LOCAL and my custom simple interceptor for @Transactional functionality.

If any one have working project with Helidon MP JPA, JTA & Hikari CP, kindly let me know. Except this, I see Helidon is a great project which consumes very less memory & with simple packaging mechanism.

**Comment by ljnelson** [comment]
Hello; I'm sorry you've had trouble in this area.  May I see a reproducer?  We have unit tests in this general area that pass successfully on every build.  It's possible we missed something.  You may also want to keep an eye on #1146.

**Comment by nagaraja0yellapu** [comment]
Here is the github repo which reproduces the issues: https://github.com/nagaraja0yellapu/helidon-jpa-jta-test . This happens if some Query runs after changing data & before throwing exception which is flushing data to db and not rolledback after.

With JDK8+
With PostgreSQL 9.3

```bash
mvn clean install
java -jar testProject.jar
```

```
create data:

http://localhost:8080/rest/testData/create?name=test111
http://localhost:8080/rest/testData/create?name=test222
http://localhost:8080/rest/testData/create?name=test333

view created data by id:
http://localhost:8080/rest/testData/getById/1
{"createdBy":"admin","createdTime":"2019-12-05T15:04:15.702Z[UTC]","description":"","id":1,"lastModifiedTime":"2019-12-05T15:04:15.702Z[UTC]","name":"test111"}


test (description field is saved even though BadRequestException is thrown)
http://localhost:8080/rest/testData/test?id=1
{"createdBy":"admin","createdTime":"2019-12-05T15:04:15.702Z[UTC]","description":"data should not save as the exception is thrown, it seems to be bug!!!","id":1,"lastModifiedTime":"2019-12-05T15:04:15.702Z[UTC]","name":"test1"}

```

**Comment by nagaraja0yellapu** [comment]
Kindly check README file for the updated instructions @ https://github.com/nagaraja0yellapu/helidon-jpa-jta-test

**Comment by ljnelson** [comment]
Thanks.  I'll be looking at this today.

Now that #1146 has been merged, I wonder if you could try your reproducer with a local build of Helidon.  To do this, clone the Helidon project and run:

    mvn clean install -DskipTests

Then, in your reproducer project, update Helidon version references to `2.0-SNAPSHOT`.

**Comment by ljnelson** [comment]
I almost forgot: you should leave the auto-commit settings up to the JPA provider.  EclipseLink, for example, needs auto commit to be supported and "on" in the vast majority of cases (this is OK since SQL is sent to the database at JPA `flush()` time, not at `persist()` or `merge()` time): 

https://github.com/eclipse-ee4j/eclipselink/blob/14e754b5159b1a28441452fa9dda45e74fb294bb/foundation/org.eclipse.persistence.core/src/main/java/org/eclipse/persistence/internal/databaseaccess/DatabasePlatform.java#L637-L650

Hibernate will adjust the auto-commit settings on the fly as needed.  So leaving the pool defaults alone is the proper thing to do.

**Comment by ljnelson** [comment]
See #1316.

**Comment by ljnelson** [comment]
Fixed with #1316.
